specs/experimental/contracts/L2/l1-fee-vault.md

```diff
@@ -0,0 +1,237 @@
+    - [Impact](#impact-2)
+  - [i01-004: Configuration parameters are immutable after deployment](#i01-004-configuration-parameters-are-immutable-after-deployment)
+    - [Impact](#impact-3)
+  - [i01-005: Only authorized withdrawals succeed](#i01-005-only-authorized-withdrawals-succeed)
+    - [Impact](#impact-4)
+- [Function Specification](#function-specification)
+  - [constructor](#constructor)
+  - [receive](#receive)
+  - [minWithdrawalAmount](#minwithdrawalamount)
+  - [recipient](#recipient)
+  - [withdrawalNetwork](#withdrawalnetwork)
+  - [withdraw](#withdraw)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The L1FeeVault accumulates the L1 portion of transaction fees paid by users on the L2 network. These fees
+represent the cost of posting transaction data to L1 for data availability. The contract enables permissionless
+withdrawal of accumulated fees to a designated recipient address once a minimum threshold is reached.
+
+## Definitions
+
+### Minimum Withdrawal Amount
+
+The minimum balance of ETH that must accumulate in the vault before a withdrawal can be triggered. This threshold
+prevents excessive withdrawal transactions for small fee amounts.
+
+### Withdrawal Network
+
+The destination network where withdrawn fees are sent. Can be either L1 (cross-domain withdrawal via message
+passing) or L2 (direct transfer on the same chain).
+
+## Assumptions
+
+### a01-001: Recipient address is valid and can receive ETH
+
+The recipient address configured at deployment must be capable of receiving ETH on the specified withdrawal
+network. For L1 recipients, the address must be able to receive ETH from the OptimismPortal. For L2 recipients,
+the address must not be a contract that reverts on ETH receipt.
+
+#### Mitigations
+
+- Deployment configuration should validate recipient addresses before deployment
+- Recipient addresses should be tested with small transfers before production use
+
+### a01-002: L2ToL1MessagePasser functions correctly for L1 withdrawals
+
+When the withdrawal network is L1, the contract depends on the L2ToL1MessagePasser predeploy to correctly
+initiate cross-domain withdrawals. The message passer must properly record withdrawal commitments and the L1
+OptimismPortal must honor these commitments.
+
+#### Mitigations
+
+- L2ToL1MessagePasser is a core protocol predeploy with extensive testing and auditing
+- Withdrawal proofs are validated on L1 through the fault proof system
+
+### a01-003: Protocol correctly credits L1 fees to this contract
+
+The execution engine must correctly calculate L1 data availability fees and credit them to the L1FeeVault
+address. Incorrect fee calculation or routing would prevent the vault from accumulating the intended fees.
+
+#### Mitigations
+
+- Fee calculation is part of the core protocol specification
+- Fee routing is handled by the execution engine's coinbase mechanism
+
+## Dependencies
```

Comment:
devinai Let's just get rid of the dependencies section for now, it's too complex